cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java

```diff
@@ -2094,10 +2096,21 @@ void testConnectorResourceMetricsScaledToZero(VertxTestContext context) {
         MeterRegistry meterRegistry = metricsProvider.meterRegistry();
         Tags tags = Tags.of("kind", KafkaConnector.RESOURCE_KIND, "namespace", namespace);
 
+        LOGGER.info("Pausing KafkaConnect reconciliations");
+        KafkaConnect pausedConnect = new KafkaConnectBuilder(kafkaConnect)
+            .editOrNewMetadata()
+                .addToAnnotations(Annotations.ANNO_STRIMZI_IO_PAUSE_RECONCILIATION, "true")
+            .endMetadata()
+            .build();
+        Crds.kafkaConnectOperation(client).inNamespace(namespace).resource(pausedConnect).update();
+        waitForConnectPaused(connectName);
+
+        LOGGER.info("Triggering reconcileAll operation");
```

Comment:
maybe I missed the long story of this bug but now we are pausing and then reconciling, doesn't it mean that the resource is not reconciled at all? How does it fix the issue?

Reply:
Here we are testing `reconcileAll` and when KC reconciliations are active they also call this method (race condition). IMO this was the source of flakiness here.